.github/workflows/nix.yml

```diff
@@ -41,14 +43,43 @@ jobs:
         extra_nix_config: |
           experimental-features = nix-command flakes
         nix_path: nixpkgs=channel:nixos-unstable
-    - if: ${{ needs.pre_job.outputs.should_skip != 'true' && env.HAS_TOKEN == 'true' }}
-      uses: cachix/cachix-action@v8
+    - if: ${{ needs.pre_job.outputs.should_skip != 'true' }}
+      uses: cachix/cachix-action@v10
       with:
         name: haskell-language-server
-        authToken: ${{ secrets.HLS_CACHIX_AUTH_TOKEN }}
+        # Disable pushing, we will do that in job `build`
+        skipPush: true
```

Comment:
is this a good idea? I expect that the `nix build` job will be failing most of the time.

Reply:
I think we could prioritize pushing the cache of the development shell in the job `build`, so that the push will take place even if `nix build` and the job fail. If I understand correctly, the cache of development shell is very small, where most of drvs are provided by https://cache.nixos.org/. And I believe that `nix develop` now works smoothly without setting up our cachix.